File: Cpp/338.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> ret;
        ret.push_back(0);
        for(int i = 1; i <= num; i++)
        {
            int bits = ret[i >> 1] + (i & 1);
            ret.push_back(bits);
        }
        return ret;
    }
};

File: Python/338.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        rList=[0 for i in range(num+1)]
        
        
        for i in range(1,num+1):
            rList[i]=rList[i>>1] + (i&1)
            
        return rList


     #beat 98%